packages/order/src/migrations/Migration20240219102530.ts

```diff
@@ -0,0 +1,462 @@
+          "shipping_address_id" text NULL,
+          "billing_address_id" text NULL,
+          "no_notification" boolean NULL,
+          "summary" jsonb NOT NULL,
+          "metadata" jsonb NULL,
+          "created_at" timestamptz NOT NULL DEFAULT now(),
+          "updated_at" timestamptz NOT NULL DEFAULT now(),
+          "deleted_at" timestamptz NULL,
+          "canceled_at" timestamptz NULL,
+          CONSTRAINT "order_pkey" PRIMARY KEY ("id")
+      );
+
+      ALTER TABLE "order"
+      ADD COLUMN if NOT exists "deleted_at" timestamptz NULL;
+
+      ALTER TABLE "order"
+      ADD COLUMN if NOT exists "original_order_id" text NULL;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_6ff7e874f01b478c115fdd462eb" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_19b0c6293443d1b464f604c3316" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_717a141f96b76d794d409f38129" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_727b872f86c7378474a8fa46147" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_5568d3b9ce9f7abeeb37511ecf2" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_c99a206eb11ad45f6b7f04f2dcc" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_cd7812c96209c5bdd48a6b858b0" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "FK_e1fcce2b18dbcdbe0a5ba9a68b8" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "REL_c99a206eb11ad45f6b7f04f2dc" CASCADE;
+
+      ALTER TABLE "order" DROP constraint if EXISTS "UQ_727b872f86c7378474a8fa46147" CASCADE;
+
+      DROP INDEX if exists "IDX_19b0c6293443d1b464f604c331";
+
+      DROP INDEX if exists "IDX_579e01fb94f4f58db480857e05";
+
+      DROP INDEX if exists "IDX_5568d3b9ce9f7abeeb37511ecf";
+
+      DROP INDEX if exists "IDX_c99a206eb11ad45f6b7f04f2dc";
+
+      DROP INDEX if exists "IDX_cd7812c96209c5bdd48a6b858b";
+
+      DROP INDEX if exists "IDX_e1fcce2b18dbcdbe0a5ba9a68b";
+
+      ${generatePostgresAlterColummnIfExistStatement(
+        "order",
+        ["fulfillment_status", "payment_status", "display_id"],
+        "DROP NOT NULL"
+      )}
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_region_id" ON "order" (
+          region_id
+      )
+      WHERE deleted_at IS NOT NULL;
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_customer_id" ON "order" (
+          customer_id
+      )
+      WHERE deleted_at IS NOT NULL;
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_original_order_id" ON "order" (
+          original_order_id
+      )
+      WHERE deleted_at IS NOT NULL;
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_customer_id" ON "order" (
+          customer_id
+      )
+      WHERE deleted_at IS NOT NULL;
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_currency_code" ON "order" (
+          currency_code
+      )
+      WHERE deleted_at IS NOT NULL;
+
+      CREATE INDEX IF NOT EXISTS "IDX_order_shipping_address_id" ON "order" (
```

Comment:
Do we need all these indices? For example, I think, for example, shipping address ID is not something you will search orders by.

Reply:
No normally, but we are defining all the graph in our joiner configs + link modules. So, from a shipping address, one can expand orders.
If in the future any index size become a concern, we can review that, for now traversing the entities from any entrypoint is a nice to have feature.

Reply:
Got it 👍

Reply:
@carlos-r-l-rodrigues one more question 😄 is it required that we have an index when doing the linking, or we do it for performance? If it is purely from a performance POV, I think if you have an index on a column that will always have only 2-3 values (eg. shipping method), Postgres might decide to not use the index so you end up with wasted storage (and it impacts write performance). Just a thought 😄